### DISCUSSION TITLE
"GetCompositeSchedule is not supported for connector 0 in Charge Point role"  why? what is the reason?

### DISCUSSION THREAD

**Discussion by mahdidadashi65** [c-jimenez/open-ocpp#215] [Q&A]
I could not pass the "Get Composite Schedule" TestCase(TC_066_CS) of Smart Charging with the new OCTT test tool and after some checks, I understand that the test tool needs us to compose connector 0, max power value with connector 1 TXprofile max value and get min value.
In the readme file, you notified that "GetCompositeSchedule is not supported for connector 0 in Charge Point role" What is the problem with this feature not being implemented?

**Comment by mahdidadashi65** [comment]
@c-jimenez can you help me?

**Reply by c-jimenez** [reply]
Sorry I didn't see your message.

The reason behind this limitation was that I was not sure how to compute the schedule on connector 0.

Maybe if you can send me your test case in details I will infere how to compute it.
You could also try to adapt the existing code for non 0 connectors and see if it works as expected.

**Comment by mahdidadashi65** [comment]
Thanks for your reply.
You are absolutely right , Until now, there was no specific document for how to test smart charging and 
an example for the correct input and output, but this problem has been solved in the new test tool.
There were two problems in the this test case :
First, the duration must be the same as the requested value.
Secondly, the limits of the maximum power or current of the connector zero should also be considered.
I solved the first problem, but unfortunately I could not do anything about the second problem.
The full text of the ،Test Case description is given in the attachment.
(https://github.com/user-attachments/assets/900463d7-3b05-43bc-afda-ca0de299be58)

**Reply by mahdidadashi65** [reply]
@c-jimenez can you help me?
